centrifuge/src/main/java/io/github/centrifugal/centrifuge/Client.java

```diff
@@ -873,7 +873,15 @@ private void handlePub(String channel, Protocol.Publication pub) {
         Subscription sub = this.getSub(channel);
         if (sub != null) {
             PublicationEvent event = new PublicationEvent();
-            event.setData(pub.getData().toByteArray());
+            byte[] pubData = pub.getData().toByteArray();
+            byte[] prevData = sub.getPrevData();
+            if (prevData != null && pub.getDelta()) {
+                try {
+                    pubData = Fossil.applyDelta(prevData, pubData);
+                } catch (Exception e) {}
```

Comment:
We should not suppress an error here – as it will mean silent message loss. I think the exception here signals about the bug to be fixed, and which should never happen in normal situation. So just letting the Exception raise seems a better approach.

Probably I am missing some cases you've encountered during implementation – let me know then.

Reply:
> the exception here signals about the bug to be fixed, and should never happen in normal situation. So just letting the Exception raise seems a better approach.

Nice and precise :+1: